kbs/src/token/mod.rs

```diff
@@ -96,7 +98,12 @@ impl TokenVerifier {
         for path in &self.extra_teekey_paths {
             if let Some(pkey_value) = claim.pointer(path) {
                 debug!("Extract tee public key from {path}");
-                return TeePubKey::deserialize(pkey_value).map_err(|_| Error::TeePubKeyParseFailed);
+                match pkey_value {
+                    Value::String(s) => {
+                        serde_json::from_str(s).map_err(Error::TeePubKeyParseFailed)?
+                    }
+                    _ => return Err(Error::TeePubKeyInvalidJwkString),
+                }
```

Comment:
Note that your new logic will assume the JWK is a string, while the original logic assumes it is a JSON object. The difference (I did not follow the JWK format but to show the difference):

Original one
```json
{
  "tee-pubkey": {
    "a": "123"
  }
}
```

change in this PR
```json
{
  "tee-pubkey": "{\"a\":\"123\"}"
}
```

Reply:
Thank you, this was a mistake in the way I was serializing my key, not in the way it was being deserialized. There may still need to be a modification there but it can be addressed in another PR. I've dropped that commit.

Reply:
Thanks, I modified the commit to allow the public key to be represented as a string, but not required to be.

Reply:
I'm trying to work around a bug in another crate that is resulting in objects to be represented as strings in `serde_json`. You're correct that this is a workaround. I'll mark this as draft for now and try to fix the problem in the producing crate.

Reply:
The serialization issue was fixed in `rust-ear` version `0.4.0`. Commit removed.